llvm/lib/Transforms/Vectorize/VPlanUtils.cpp

```diff
@@ -113,7 +113,12 @@ bool vputils::isUniformAcrossVFsAndUFs(VPValue *V) {
                all_of(R->operands(),
                       [](VPValue *Op) { return isUniformAcrossVFsAndUFs(Op); });
       })
-      .Case<VPScalarCastRecipe, VPWidenCastRecipe>([](const auto *R) {
+      .Case<VPInstruction>([](const auto *VPI) {
+        return Instruction::isCast(VPI->getOpcode())
+                   ? all_of(VPI->operands(), isUniformAcrossVFsAndUFs)
+                   : false;
```

Comment:
Nit, I think this is the same as

```suggestion
        return Instruction::isCast(VPI->getOpcode()) && all_of(VPI->operands(), isUniformAcrossVFsAndUFs);
```

Reply:
Done, thanks!

Reply:
Ah yes, just thought passing to all_of directly, but updated now, thanks